apps/staking/hooks/parseContracts.ts

```diff
@@ -0,0 +1,173 @@
+import { ARBITRUM_EVENT, CONTRIBUTION_CONTRACT_STATUS } from '@session/staking-api-js/enums';
+import type { ContributionContract } from '@session/staking-api-js/schema';
+import { areHexesEqual } from '@session/util-crypto/string';
+import type { Address } from 'viem';
+import { getTotalStakedAmountForAddress } from '../components/getTotalStakedAmountForAddress';
+import logger from '../lib/logger';
+import { sortEvents } from './parseEvents';
+import { sortingReservedContractsDesc, sortingTotalStakedDesc } from './parseStakes';
+
+const contractStateSortOrderIfOperator = {
+  [CONTRIBUTION_CONTRACT_STATUS.WaitForFinalized]: 1,
+  [CONTRIBUTION_CONTRACT_STATUS.WaitForOperatorContrib]: 2,
+  [CONTRIBUTION_CONTRACT_STATUS.OpenForPublicContrib]: 3,
+  [CONTRIBUTION_CONTRACT_STATUS.Finalized]: 4,
+};
+
+/**
+ * If the connected wallet is the contract operator, then the contracts are sorted by {@link contractStateSortOrderIfOperator}
+ *
+ * If the state is the same OR the connected wallet isn't the contract operator, then the contract are sorted by the total staked amount descending
+ * then by the operator fee ascending
+ */
+export function sortContracts(a: ContributionContract, b: ContributionContract, address?: Address) {
+  const operatorA = areHexesEqual(a.operator_address, address);
+  const operatorB = areHexesEqual(b.operator_address, address);
+
+  const priorityA = operatorA ? (contractStateSortOrderIfOperator[a.status] ?? 999) : 999;
```

Comment:
Should we not use Number.POSITIVE_INFINITY here? Unless there is a cap at 999?

Reply:
Good catch, changed

Reply:
chore task, change made in https://github.com/session-foundation/websites/pull/46/commits/8ad447adf392b150d1ce364f3d6192111ed28b2b